app/src/main/java/com/example/poste/models/Post.java

```diff
@@ -67,6 +76,24 @@ public String getUrl() {
      */
     public String getId() { return id; }
 
+    /**
+     * Deletes this post from the data model and backend
+     */
+    public void deletePost(Folder parentFolder) {
```

Comment:
I would contend this should be a method on Folder, rather than on Post. A Post is deleted from a folder, and a folder manages its children (the posts within)

Reply:
I was a little unsure on this. because app-side. Folders contain posts and a post would be deleted be simply removing it from the folder's list.
But I saw that on the backend, the post has control. From what I could tell the folders in the data base do not even know what posts they have, but it is the post that knows to which folder it belongs. SO i was trying to mimic that post centric management here. But I'm fine doing it the other way too. I just wanted to explain why I did it like this.

Reply:
The backend is a slightly different beast. The database requires ForeignKey relations: models relate to others via ForeignKey (unsure if you've taken a database course or not; Django uses ORM, which is Object-Relational-Mapping, so while we don't directly interact with the database, Django does it for us -- and it uses SQL standards, so concepts relate)

A Post has a ForeignKey to a Folder, which is the way of stating a Post belongs to a Folder; in this way, a Post has one Folder, but a Folder can have many posts. Django *does* have a way to go "the other direction" though:

```
folder = Folder.objects.get(id=some_id)
posts_in_folder = folder.post_set.all()
```

In reality though, after coding post deletion, it seemed logical to place the methods on User. Your mileage may vary.